<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns="http://www.tibco.com/pe/EngineTypes" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:wsdlPfx="http://www.tibco.com/CommonLE2/namespace/ExceptionManagement/GUIServices" xmlns:ns5="http://www.tibco.com/CommonLE2/namespace/Private/Exception" xmlns:ns12="http://www.tibco.com/CommonLE/namespace/Common" xmlns:pfx="http://www.tibco.com/pe/DeployedVarsType" xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:ns2="http://www.tibco.com/CommonLE2/namespace/Administration/GUIServices/Applications" xmlns:ns1="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:ns4="http://www.tibco.com/CommonLE2/Private/Exception" xmlns:ns3="http://www.tibco.com/CommonLE2/Public/Common" xmlns:pfx4="http://www.tibco.com/CommonLE2/namespace/private/Common" xmlns:pfx3="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/ExceptionManagement" xmlns:pfx2="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/ExceptionManagement/ExceptionDetailQuery">
    <wsdl:import namespace="http://www.tibco.com/CommonLE2/namespace/ExceptionManagement/GUIServices" location="/SharedResources/WSDL/GUI/ExceptionManagement/intfResolveExceptions.wsdl"/>
    <pd:name>ExceptionManagement/GUIServices/wsRetrieveExceptionDetailWithRendering.process</pd:name>
    <pd:startName>Start</pd:startName>
    <pd:startType wsMsgRef="wsdlPfx:ExceptionDetailQuery"/>
    <pd:startX>83</pd:startX>
    <pd:startY>55</pd:startY>
    <pd:returnBindings>
        <wsdlPfx:ExceptionDetail>
            <parameters>
                <xsl:copy-of select="$Retrieve-Exception-Case-Detail-With-Rendering/pfx3:ExceptionDetail"/>
            </parameters>
        </wsdlPfx:ExceptionDetail>
    </pd:returnBindings>
    <pd:endName>End</pd:endName>
    <pd:endType wsMsgRef="wsdlPfx:ExceptionDetail"/>
    <pd:endX>378</pd:endX>
    <pd:endY>55</pd:endY>
    <pd:errorSchemas>
        <Schema1 wsMsgRef="wsdlPfx:Error"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1284614219735</pd:targetNamespace>
    <pd:activity name="Retrieve Exception Case Detail With Rendering">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>207</pd:x>
        <pd:y>69</pd:y>
        <config>
            <processName>/ExceptionManagement/Logic/RetrieveExceptionDetailwithRendering.process</processName>
            <processNameXPath/>
        </config>
        <pd:inputBindings>
            <ns5:ExceptionDetailQuery>
                <ns5:ExceptionID>
                    <xsl:value-of select="$Start/wsdlPfx:ExceptionDetailQuery/parameters/pfx2:ExceptionDetailQuery/pfx2:ExceptionID"/>
                </ns5:ExceptionID>
            </ns5:ExceptionDetailQuery>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Catch All Errors">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>85</pd:x>
        <pd:y>146</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <catchAll>true</catchAll>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Generate Internal Error">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>402</pd:x>
        <pd:y>153</pd:y>
        <config>
            <faultName>Schema1</faultName>
        </config>
        <pd:inputBindings>
            <ns1:ActivityInput>
                <xsl:if test="$_error/ns:ErrorReport/Msg">
                    <message>
                        <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                    </message>
                </xsl:if>
                <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                    <messageCode>
                        <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                    </messageCode>
                </xsl:if>
                <data>
                    <wsdlPfx:Error>
                        <parameters>
                            <ns3:Error>
                                <xsl:if test="$_error/ns:ErrorReport/Msg">
                                    <ns3:Message>
                                        <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                                    </ns3:Message>
                                </xsl:if>
                                <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                                    <ns3:Code>
                                        <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                                    </ns3:Code>
                                </xsl:if>
                            </ns3:Error>
                        </parameters>
                    </wsdlPfx:Error>
                </data>
            </ns1:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Local Exception Handler">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>250</pd:x>
        <pd:y>154</pd:y>
        <config>
            <processName>/Common/Local Exception Handler.process</processName>
        </config>
        <pd:inputBindings>
            <ns12:Exception>
                <ErrorMessage>
                    <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                </ErrorMessage>
                <ErrorCode>
                    <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                </ErrorCode>
                <ns12:Error>
                    <xsl:for-each xmlns:xsl="http://www.w3.org/1999/XSL/Transform" select="$_error/ns:ErrorReport">
                        <ns12:ErrorReport>
                            <StackTrace>
                                <xsl:value-of select="StackTrace"/>
                            </StackTrace>
                            <Msg>
                                <xsl:value-of select="Msg"/>
                            </Msg>
                            <FullClass>
                                <xsl:value-of select="FullClass"/>
                            </FullClass>
                            <Class>
                                <xsl:value-of select="Class"/>
                            </Class>
                            <ProcessStack>
                                <xsl:value-of select="ProcessStack"/>
                            </ProcessStack>
                            <MsgCode>
                                <xsl:value-of select="MsgCode"/>
                            </MsgCode>
                        </ns12:ErrorReport>
                    </xsl:for-each>
                </ns12:Error>
            </ns12:Exception>
        </pd:inputBindings>
    </pd:activity>
    <pd:transition>
        <pd:from>Start</pd:from>
        <pd:to>Retrieve Exception Case Detail With Rendering</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Retrieve Exception Case Detail With Rendering</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch All Errors</pd:from>
        <pd:to>Local Exception Handler</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Local Exception Handler</pd:from>
        <pd:to>Generate Internal Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>